File: 01-business-sub-system/src/main/java/com/github/yingzhuo/datawarehouse/businesssubsys/dao/CartItemDao.scala
package com.github.yingzhuo.datawarehouse.businesssubsys.dao

import com.github.yingzhuo.datawarehouse.businesssubsys.domain.CartItem
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
trait CartItemDao extends JpaRepository[CartItem, String] {
}

File: 01-business-sub-system/src/main/java/com/github/yingzhuo/datawarehouse/businesssubsys/dao/CartItemDao.java
package com.github.yingzhuo.datawarehouse.businesssubsys.dao;

import com.github.yingzhuo.datawarehouse.businesssubsys.domain.CartItem;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CartItemDao extends JpaRepository<CartItem, String> {

    public CartItem findByUserIdAndCommodityId(String userId, String commodityId);

    public List<CartItem> findByUserId(String userId);

}
